### PR TITLE
fix: update service worker script and remove index.js cache inlining in html

### PR DESCRIPTION
**Pull request by igorbrasileiro**
The sw.js and index.js were being cached.
The fix is inlining the index.js and updating service work.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -78,8 +78,11 @@ const registerServiceWorker = async () => {
     const registration = await navigator.serviceWorker.register("sw.js");
 
     if (registration.active) {
+      // should await?
+      await registration.update();
       waiting.resolve();
     } else if (registration.waiting || registration.installing) {
+      console.log("creating");
```
Comment (tlgimenes) [inline index.js:85]: 👀

File: sw.js [hunk 2]
```diff
@@ -5,7 +5,21 @@ self.addEventListener("install", (_event) => {
 });
 
 self.addEventListener("activate", (event) => {
-  event.waitUntil(clients.claim());
+  event.waitUntil(
```
Comment (tlgimenes) [inline sw.js]: this is not necessary. Rollback to clients.claim

File: sw.js [hunk 3]
```diff
@@ -239,6 +253,7 @@ async function maybeTranspileResponse(request, response) {
 }
 
 async function route(request) {
+  console.log("route");
```
Comment (tlgimenes) [inline sw.js]: 👀

**Review by tlgimenes — COMMENTED** [review]
it's missing the part where we should not cache the html. Also, now I realized why I was not inlining index.js. Because it's used on 404.html too. Maybe we should use a different approach

**Comment by igorbrasileiro** [comment]
> it's missing the part where we should not cache the html. Also, now I realized why I was not inlining index.js. Because it's used on 404.html too. Maybe we should use a different approach

The HTML used isn't beign cached.
image

Also, the pages aren't beign handled by the route, I mean via fetch event. Do you have a different approach to catch their requests?

**Pull request closed without merging**